client/my-sites/domains/domain-search/index.jsx

```diff
@@ -308,6 +341,7 @@ export default connect(
 			isSiteOnMonthlyPlan: isSiteOnMonthlyPlan( state, siteId ),
 			productsList: getProductsList( state ),
 			userCanPurchaseGSuite: canUserPurchaseGSuite( state ),
+			stateObject: state,
```

Comment:
Passing the entire state object to a component is an antipattern, because it will re-render on any state change. This is a good place to use the `isDomainSidebarExperimentUser` selector, which will limit the area of state object interest for this component.